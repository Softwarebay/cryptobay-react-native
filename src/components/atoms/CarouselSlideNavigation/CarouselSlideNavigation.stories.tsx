import { number } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';

import StorybookView from '../../../../storybook/StorybookView';

import { CarouselSlideNavigation } from './CarouselSlideNavigation';

storiesOf('Atoms/CarouselSlideNavigation', module)
  .addDecorator(getStory => <StorybookView>{getStory()}</StorybookView>)
  .add('Carousel Slide Navigation', () => (
    <CarouselSlideNavigation slide={number('slide', 1) as 1 | 2 | 3} />
  ));
