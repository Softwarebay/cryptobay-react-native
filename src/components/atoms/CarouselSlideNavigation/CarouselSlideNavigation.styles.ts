import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  carouselSlideNavigation: {
    flexDirection: 'row',
  },
  dot: {
    width: 8,
    height: 8,
    borderRadius: 8,
    marginHorizontal: 4,
  },
  outdatedSlideDot: {
    backgroundColor: '#4C516B',
  },
  currentSlideDot: {
    backgroundColor: '#FEBF32',
  },
});
