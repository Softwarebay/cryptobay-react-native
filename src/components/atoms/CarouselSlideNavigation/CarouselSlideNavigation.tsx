import React from 'react';
import { View } from 'react-native';

import { styles } from './CarouselSlideNavigation.styles';

type CarouselSlideNavigationProps = {
  slide: 1 | 2 | 3;
};

export const CarouselSlideNavigation: React.FC<
  CarouselSlideNavigationProps
> = ({ slide }: CarouselSlideNavigationProps) => {
  const isCurrentSlide = (slideNumber: number) =>
    slideNumber === slide ? styles.currentSlideDot : styles.outdatedSlideDot;

  return (
    <View style={styles.carouselSlideNavigation}>
      <View
        testID="slide-1-navigation-dot"
        style={[isCurrentSlide(1), styles.dot]}
      />
      <View
        testID="slide-2-navigation-dot"
        style={[isCurrentSlide(2), styles.dot]}
      />
      <View
        testID="slide-3-navigation-dot"
        style={[isCurrentSlide(3), styles.dot]}
      />
    </View>
  );
};
