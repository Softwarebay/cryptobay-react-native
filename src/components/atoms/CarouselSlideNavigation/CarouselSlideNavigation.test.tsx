import { render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { CarouselSlideNavigation } from './CarouselSlideNavigation';

describe('CarouselSlideNavigation Atom', () => {
  test('Component Snapshot', () => {
    const tree = renderer
      .create(<CarouselSlideNavigation slide={1} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure dots show current slide correctly', () => {
    const { rerender } = render(<CarouselSlideNavigation slide={1} />);
    const sliderDot = (slideNumber: number) =>
      screen.getByTestId(`slide-${slideNumber}-navigation-dot`);
    expect(sliderDot(1)).toHaveStyle({ backgroundColor: '#FEBF32' });
    expect(sliderDot(2)).toHaveStyle({ backgroundColor: '#4C516B' });
    expect(sliderDot(3)).toHaveStyle({ backgroundColor: '#4C516B' });
    rerender(<CarouselSlideNavigation slide={2} />);
    expect(sliderDot(1)).toHaveStyle({ backgroundColor: '#4C516B' });
    expect(sliderDot(2)).toHaveStyle({ backgroundColor: '#FEBF32' });
    expect(sliderDot(3)).toHaveStyle({ backgroundColor: '#4C516B' });
    rerender(<CarouselSlideNavigation slide={3} />);
    expect(sliderDot(1)).toHaveStyle({ backgroundColor: '#4C516B' });
    expect(sliderDot(2)).toHaveStyle({ backgroundColor: '#4C516B' });
    expect(sliderDot(3)).toHaveStyle({ backgroundColor: '#FEBF32' });
  });
});
