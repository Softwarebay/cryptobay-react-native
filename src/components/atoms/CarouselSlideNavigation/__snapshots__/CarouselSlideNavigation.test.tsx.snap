// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselSlideNavigation Atom Component Snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FEBF32",
        },
        Object {
          "borderRadius": 8,
          "height": 8,
          "marginHorizontal": 4,
          "width": 8,
        },
      ]
    }
    testID="slide-1-navigation-dot"
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#4C516B",
        },
        Object {
          "borderRadius": 8,
          "height": 8,
          "marginHorizontal": 4,
          "width": 8,
        },
      ]
    }
    testID="slide-2-navigation-dot"
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#4C516B",
        },
        Object {
          "borderRadius": 8,
          "height": 8,
          "marginHorizontal": 4,
          "width": 8,
        },
      ]
    }
    testID="slide-3-navigation-dot"
  />
</View>
`;
